### PR TITLE
Fixes for Chef 15

### PR DESCRIPTION
**Pull request by EMH333**
There were some workarounds needed for idempotency issues but the only major change is removing CentOS 6 support which cleaned up a lot of the code. This will be a breaking change requiring a major version increase.

Also note that I removed the explicit starting of services like `ganeti-mond` as it seems that the primary ganeti service will automatically start them once they are ready and this solves some of the services complaining about non-existent keys

### CODE REVIEW

**Comment by EMH333** [comment]
[Chef 15: ganeti](https://trello.com/c/X29dwqKu/1774-chef-15-ganeti)

File: recipes/default.rb [hunk 1]
```diff
@@ -19,16 +19,20 @@
 include_recipe 'yum-epel' if platform_family?('rhel')
 
 yum_repository 'ganeti' do
-  node['ganeti']['yum'].each do |key, value|
-    send(key.to_sym, value)
-  end
+  description  node['ganeti']['yum']['description']
+  baseurl node['ganeti']['yum']['url']
+  gpgkey node['ganeti']['yum']['gpgkey']
+  gpgcheck node['ganeti']['yum']['gpgcheck']
```
Comment (ramereth) [inline recipes/default.rb:25]: Why was this changed?

File: resources/instance_image_hook.rb [hunk 2]
```diff
@@ -8,15 +8,13 @@
 
 action :enable do
   file ::File.join(config_dir, 'hooks', new_resource.name) do
-    mode 0755
-    action :touch
```
Comment (ramereth) [inline resources/instance_image_hook.rb:12]: This action needs to stay as it's not the default action. Same for below.

File: spec/unit/recipes/default_spec.rb [hunk 3]
```diff
@@ -34,50 +33,13 @@
...
-          ganeti-wconfd
-        ).each do |ganeti_service|
-          it do
-            expect(chef_run).to enable_service(ganeti_service).with(supports: { status: true, restart: true })
-          end
-          it do
-            expect(chef_run).to start_service(ganeti_service).with(supports: { status: true, restart: true })
-          end
-        end
-      end
```
Comment (ramereth) [inline spec/unit/recipes/default_spec.rb:80]: These tests should remain, but the context is no longer needed I think.

File: resources/instance_image_hook.rb (hunk 2 above)
Comment (EMH333) [reply resources/instance_image_hook.rb:12]: I was running into idempotency issues if the action was touch for multi-converge

```
       Running handlers:
       First chef run should have reached a converged state.
       Resources updated in a second chef-client run:
       - instance_image_hook[create]
       - instance_image_hook[grub]
       - instance_image_hook[foo]
       - instance_image_hook[disable]
       - instance_image_hook[source]
```

It looks like touch is updating the utime on the files
```
       * instance_image_hook[source] action create
         * file[/etc/ganeti/instance-image/hooks/source] action touch
           - update utime on file /etc/ganeti/instance-image/hooks/source
```

Comment (ramereth) [reply resources/instance_image_hook.rb:12]: Hmm, you're right. I'll double check this works as expected when I run a local test.

File: spec/unit/recipes/default_spec.rb (hunk 3 above)
Comment (EMH333) [reply spec/unit/recipes/default_spec.rb:80]: I believe these tests already exist below, the CentOS 6 cleanup resulted in removing the configuration path where `/etc/init.d/ganeti` exists. Given those checks were removed from the recipe, I just adjusted some of the other tests in the main context (specifically the ganeti service name and not starting the helper services) 
https://github.com/osuosl-cookbooks/ganeti/blob/c1785f917560add22aec6db3327954ffb4b58ad4/spec/unit/recipes/default_spec.rb#L90-L116 to reflect what should always happen now that CentOS6 is gone.

File: recipes/default.rb (hunk 1 above)
Comment (EMH333) [reply recipes/default.rb]: Good question! :) just reverted

**Comment by EMH333** [comment]
!bump major *

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a major-level version bump to v3.0.0.  Have a nice day!
